code/game/mecha/mech_fabricator.dm

```diff
@@ -172,34 +239,26 @@
 	atom_say("[I] is complete.")
 	being_built = null
 
-	updateUsrDialog()
+	SSnanoui.update_uis(src)
 	return TRUE
 
-/obj/machinery/mecha_part_fabricator/proc/update_queue_on_page()
-	send_byjax(usr,"mecha_fabricator.browser","queue",list_queue())
-	return
-
-/obj/machinery/mecha_part_fabricator/proc/add_part_set_to_queue(set_name)
-	if(set_name in part_sets)
-		for(var/v in files.known_designs)
-			var/datum/design/D = files.known_designs[v]
-			if(D.build_type & fabricator_type)
-				if(set_name in D.category)
-					add_to_queue(D)
-
-/obj/machinery/mecha_part_fabricator/proc/add_to_queue(D)
+// Add a single part to the queue
+/obj/machinery/mecha_part_fabricator/proc/add_to_queue(datum/design/D)
 	if(!istype(queue))
 		queue = list()
-	if(D)
+	if(D && istype(D))

```

Comment:
This is the same as `if(istype(D))`, right?

Reply:
Right, but isn't `if(D != null && istype(D))` equivalent to `if(istype(D))` ?

Reply:
I'm honestly not sure what istype does if you do `istype(null)`. I notice in most cases these types of safety checks include both of these, similar to how I have it written.